include/boost/geometry/geometries/PolyhedralSurface.hpp

```diff
@@ -0,0 +1,103 @@
+#ifndef BOOST_GEOMETRY_GEOMETRIES_POLYHEDRALSURFACE_HPP
+#define BOOST_GEOMETRY_GEOMETRIES_POLYHEDRALSURFACE_HPP
+
+#include <memory>
+#include <vector>
+
+#include <boost/concept/assert.hpp>
+
+#include <boost/geometry/geometries/concepts/point_concept.hpp>
+#include <boost/geometry/geometries/ring.hpp>
+#include <boost/geometry/core/tag.hpp>
+#include <boost/geometry/core/tags.hpp>
+
+#include <boost/config.hpp>
+#ifndef BOOST_NO_CXX11_HDR_INITIALIZER_LIST
+#include <initializer_list>
+#endif
+
+namespace boost { namespace geometry 
+{
+
+namespace model
+{
+
+
+template 
+<
+    typename Point,
+    bool ClockWise = true, bool Closed = true,
+    template<typename, typename> class PointList = std::vector,
+    template<typename, typename> class RingList = std::vector,
+    template<typename> class PointAlloc = std::allocator,
+    template<typename> class RingAlloc = std::allocator
+>
+class PolyhedralSurface : public PointList<ring<Point>, RingAlloc<ring<Point> > >
+{
+	BOOST_CONCEPT_ASSERT( (concepts::Point<Point>) );
+
+public :
+
+	//typedef Point point_type;
+    using point_type = Point;
+    using ring_type =  ring<Point, ClockWise, Closed, PointList, PointAlloc>;
```

Comment:
So you make use of the `ring` model. That's basically fine, but can't you simplify this surface a lot by just specifying the ring type used?

So then it would look like:
```
template
<
    typename Ring,
    template<typename, typename> class Container = std::vector,
    template<typename> class Allocator = std::allocator
>
class polyhedral_surface : public Container<Ring, Allocator<Ring> >
...
```

And you can use it like:
```
using point_type = model::point<default_test_type, 2, bg::cs::cartesian>;
using ring_type = model::ring<Point, true, true>;
using ph = model::polyhedral_surface<ring_type>;
...

That way you don't have to bother about open, closed, it's all defined by the ring concept itself, and you've only one list of rings defining your 3d surface.
```

Reply:
Ok, I got this. I will have to specify the ring type rather than using a default ring.